ledger/metrics/src/test/scala/com/daml/metrics/InstrumentedSourceSpec.scala

```diff
@@ -47,6 +54,37 @@ final class InstrumentedSourceSpec extends AsyncFlatSpec with Matchers with Akka
     }
   }
 
+  it should "correctly measure queue delay" in {
+    val capacityCounter = new Counter()
+    val maxBuffered = new InstrumentedSourceSpec.MaxValueCounter()
+    val delayTimer = new Timer()
+    val bufferSize = 2
+
+    val (source, sink) =
+      InstrumentedSource
+        .queue[Int](16, OverflowStrategy.backpressure, capacityCounter, maxBuffered, delayTimer)
+        .map { x =>
+          // Sleep to delay the processing of the next element.
+          Thread.sleep(5)
+          x
+        }
```

Comment:
Perhaps use the scheduler here rather than blocking a Future thread?

```suggestion
        .mapAsync(1) { x =>
          scheduler.scheduleOnce(5.seconds)(Future(x))
        }
```

Reply:
Thanks, good suggestion. I'll give it a try. (It's 5 millis though).

Reply:
scheduler didn't have the right methods, but found this instead: https://doc.akka.io/docs/akka/current/futures.html.

Reply:
Oh, my bad. Totally fine with the `Thread.sleep` then.